src/main/java/io/split/android/client/impressions/HttpImpressionsSender.java

```diff
@@ -28,40 +29,31 @@ public HttpImpressionsSender(CloseableHttpClient client, String eventsEndpoint,
     }
 
     @Override
-    public void post(List<TestImpressions> impressions) {
+    public boolean post(List<TestImpressions> impressions) {
+        if (impressions == null || impressions.isEmpty()) {
+            return true;
```

Comment:
why true and not false meaning nothing got posted?

Reply:
If it is succeeded it will delete the List<TestImpressions> from the storage, it is not, it will rewrite the chunk of impressions for later process. 
So if we return false the empty list of impressions will be stored again for later process, meaning it will get here and fail again since it is empty.
Let me change it so the storage one if there are no impressions to store won't do anything